elephant/spike_train_correlation.py

```diff
@@ -749,3 +751,68 @@ def run_T(spiketrain):
 
 
 sttc = spike_time_tiling_coefficient
+
+
+def spike_train_timescale(binned_st, tau_max):
+    """
+    Calculates the auto-correlation time of a binned spike train.
+    Uses the definition of the auto-correlation time proposed in [1, Eq. (6)]:
+
+    .. math::
+        \\tau_\mathrm{corr} = \\int_{-\\tau_\mathrm{max}}^{\\tau_\mathrm{max}}\
+            \left[ \\frac{\hat{C}(\\tau)}{\hat{C}(0)} \\right]^2 d\\tau
+
+    where :math:`\hat{C}(\\tau) = C(\\tau)-\\nu\delta(\\tau)` denotes
+    the auto-correlation function excluding the Dirac delta at zero timelag.
+
+    Parameters
+    ----------
+    binned_st : elephant.conversion.BinnedSpikeTrain
+        A binned spike train containing the spike train to be evaluated.
+    tau_max : quantities.Quantity
+        Maximal integration time of the auto-correlation function.
+
+    Returns
+    -------
+    timescale : quantities.Quantity
+        The auto-correlation time of the binned spiketrain.
+
+    Notes
+    -----
+    * :math:`\\tau_\mathrm{max}` is a critical parameter: numerical estimates
+      of the auto-correlation functions are inherently noisy. Due to the
+      square in the definition above, this noise is integrated. Thus, it is
+      necessary to introduce a cutoff for the numerical integration - this
+      cutoff should be neither smaller than the true auto-correlation time
+      nor much bigger.
+    * The binsize of binned_st is another critical parameter as it defines the
+      discretisation of the integral :math:`d\\tau`. If it is too big, the
+      numerical approximation of the integral is inaccurate.
+
+    References
+    ----------
+    [1] Wieland, S., Bernardi, D., Schwalger, T., & Lindner, B. (2015).
+        Slow fluctuations in recurrent networks of spiking neurons.
+        Physical Review E, 92(4), 040901.
+    """
+    T = binned_st.t_stop - binned_st.t_start
+    binsize = binned_st.binsize
+
+    rate = np.sum(binned_st.to_array()) / T
+
+    tau_max.units = binsize.units
+    tau_max_bins = int(np.round(tau_max/binsize))
```

Comment:
`np.round` is useless here

Reply:
or did you mean not taking the integer part but rather rounding to the nearest integer? if so, this behavior should be avoided - use `math.ceil`. no integer casting is required afterward.

Reply:
the problem is something like this: `int(6.6e0/3.3e-2)` which yields `199` instead of `200` due to numerical inaccuracies in float division. `ceil` would fix this example, but not if the floating division slightly overestimates the result. the only solution I see is rounding first and then taking the integer.

Reply:
`math.ceil(3.6/1.2e-3)` for example yields `3001`

Reply:
What about this?
```
    assert (tau_max/binsize).units == pq.dimensionless
    tau_max_bins = int(np.round((tau_max/binsize).simplified.magnitude))
```